Update Blazor Identity scaffolding guidance (#33134)
dotnet/AspNetCore.Docs: aspnetcore/security/authentication/scaffold-identity.md

:::moniker range=">= aspnetcore-8.0"

## Blazor Identity scaffolding

ASP.NET Core Identity scaffolding adds [ASP.NET Core Identity](xref:security/authentication/identity) to Blazor Web Apps and Blazor Server apps. Identity Razor components added to the app can be customized to change their behavior.

Although the scaffolder generates the necessary C# code to scaffold Identity into the app, you must update the project's database with an Entity Framework (EF) Core database migration to complete the process. This article explains the steps required to migrate a database.

> We recommend using a source control system that shows file differences and allows you to back out of changes. Inspect the changes after running the Identity scaffolder.

Reviewer: Just a thought

```suggestion
Inspect the changes after running the Identity scaffolder. We recommend using a GitHub or another source control system that shows file changes and provides a way to revert changes.
```

Services are required when using [two-factor authentication (2FA)](xref:blazor/security/server/qrcodes-for-authenticator-apps), [account confirmation and password recovery](xref:blazor/security/server/account-confirmation-and-password-recovery), and other security features with Identity. Services or service stubs aren't generated when scaffolding Identity. Services to enable these features must be added manually.

## Razor Pages and MVC Identity scaffolding

> ASP.NET Core provides [ASP.NET Core Identity](xref:security/authentication/identity) as a [Razor class library (RCL)](xref:razor-pages/ui-class). Applications that include Identity can apply the scaffolder to selectively add the source code contained in the Identity RCL. You might want to generate source code so you can modify the code and change the behavior. For example, you could instruct the scaffolder to generate the code used in registration. Generated code takes precedence over the same code in the Identity RCL. To gain full control of the UI and not use the default RCL, see the section [Create full Identity UI source](#full).

Reviewer: Your's is fine, feel free to ignore.

```suggestion
ASP.NET Core provides [ASP.NET Core Identity](xref:security/authentication/identity) as a [Razor class library (RCL)](xref:razor-pages/ui-class). Applications that include Identity can apply the scaffolder to selectively add the source code contained in the Identity RCL. You might want to generate source code so you can modify the code and change the behavior. For example, you could instruct the scaffolder to generate the code used in registration. Customized Identity code overrides the default implementation provided by the Identity Razor Class Library (RCL). To gain full control of the UI and not use the default RCL, see the section [Create full Identity UI source](#full).
```

Or something like Scaffoled Identity code supersedes the RCL defaults.

Typically, apps that were created with individual accounts should ***not*** create a new data context.

## Scaffold Identity into a Blazor project

*This section applies to Blazor Web Apps and Blazor Server apps.*

Run the Identity scaffolder:

# [Visual Studio](#tab/visual-studio)

* From **Solution Explorer**, right-click on the project > **Add** > **New Scaffolded Item**.
* From the left pane of the **Add New Scaffolded Item** dialog, select **Identity**. Select **Blazor Identity** in the center pane. Select the **Add** button.
* In the **Add Blazor Identity** dialog:
  * Select or add with the plus (**+**) button the database context class (**DbContext class**).
  * Select the database provider (**Database provider**), which defaults to SQL Server.
  * Select or add with the plus (**+**) button the user class (**User class**).
  * Select the **Add** button.

# [.NET CLI](#tab/net-cli)

To add the required NuGet packages and tools, execute the following .NET CLI commands in a command shell opened to the project's root folder.

> Paste all of the following commands at the prompt (`>`) of the command shell. When you paste multiple commands, a warning appears stating that multiple commands will execute. Dismiss the warning and proceed with the paste operation.

Reviewer: Do you dismiss or ignore? IIRC, there is nothing to dismiss. When I read "Dismiss", I'm thinking I must take action other.

Author: I can fix this ... the actual button text to get rid of the warning is **Paste anyway**, so I'll call that button out by name ...

```suggestion
Paste all of the following commands at the prompt (`>`) of the command shell. When you paste multiple commands, a warning appears stating that multiple commands will execute. Select the **Paste anyway** button.
```

> When you paste multiple commands, all of the commands execute except the last one. The last command doesn't execute until you press <kbd>Enter</kbd> on the keyboard.

Reviewer: I think execute is flagged by poly woke check?

```suggestion
When you paste multiple commands, all but the last one run. Press <kbd>Enter</kbd> to run the last command.
```
